tests/core/base/test_prompt.py

```diff
@@ -229,3 +234,115 @@ class BookRecommendationPrompt(BasePrompt): ...
     def fn() -> None: ...
 
     assert hasattr(fn, "_metadata") and fn._metadata == {"tags": {"version:0001"}}  # pyright: ignore [reportFunctionMemberAccess]
+
+
+def test_prompt_template_str_return():
+    @prompt_template
+    def recommend_book(genre: str) -> str:
+        return f"recommend a {genre} book"
```

Comment:
I don't know if this is your preference, but here it would be correct to define ` -> str`, create a function and have the `@prompt_template` decorator change the return value of the function.
This will allow pyright and the IDE to do type checking and completion correctly.

Reply:
hmm, ideally i want `def recommend_book(genre: str) -> Messages` in all cases unless we think this is more clear?

Reply:
I understand your intention, but wouldn't it be confusing from a type checking and usability standpoint for an undecorated function to return the decorated form (`Messages`)?

Reply:
what do you mean by decorated form? I view `Messages` as the undecorated form, and `list[BaseMessageParam]` as the decorated return type